src/stcal/ramp_fitting/ols_cas22/_fixed.pyx

```diff
@@ -0,0 +1,213 @@
+"""
+Define the data which is fixed for all pixels to compute the CAS22 algorithm with
+    jump detection
+
+Objects
+-------
+FixedValues : class
+    Class to contain the data fixed for all pixels and commonly referenced
+    universal values for jump detection
+
+Functions
+---------
+fixed_values_from_metadata : function
+    Fast constructor for FixedValues from the read pattern metadata
+"""
+import numpy as np
+cimport numpy as np
+cimport cython
+
+from stcal.ramp_fitting.ols_cas22._core cimport Thresh, ReadPatternMetadata, Diff
+from stcal.ramp_fitting.ols_cas22._fixed cimport FixedValues
+
+cdef class FixedValues:
+    """
+    Class to contain all the values which are fixed for all pixels for a given
+    read pattern.
+        This class is used to pre-compute these values once so that they maybe
+        reused for all pixels.  This is done for performance reasons.
+
+    Parameters
+    ----------
+    use_jump : bool
+        flag to indicate whether to use jump detection (user input)
+
+    data : ReadPatternMetadata
+        Metadata struct created from a read pattern
+
+    threshold : Thresh
+        Parameterization struct for threshold function
+
+    t_bar_diffs : float[:, :]
+        These are the differences of t_bar used for jump detection.
+            single differences of t_bar:
+                t_bar_diffs[Diff.single, :] = (t_bar[i+1] - t_bar[i])
+            double differences of t_bar:
+                t_bar_diffs[Diff.double, :] = (t_bar[i+2] - t_bar[i])
+    t_bar_diff_sqrs : float[:, :]
+        These are the squared differnences of t_bar used for jump detection.
+            single differences of t_bar:
+                t_bar_diff_sqrs[Diff.single, :] = (t_bar[i+1] - t_bar[i])**2
+            double differences of t_bar:
+                t_bar_diff_sqrs[Diff.double, :] = (t_bar[i+2] - t_bar[i])**2
+    read_recip_coeffs : float[:, :]
+        Coefficients for the read noise portion of the variance used to compute
+        the jump detection statistics. These are formed from the reciprocal sum
+        of the number of reads.
+            single sum of reciprocal n_reads:
+                recip[Diff.single, :] = ((1/n_reads[i+1]) + (1/n_reads[i]))
+            double sum of reciprocal n_reads:
+                recip[Diff.double, :] = ((1/n_reads[i+2]) + (1/n_reads[i]))
+    var_slope_coeffs : float[:, :]
+        Coefficients for the slope portion of the variance used to compute the
+        jump detection statistics, which happend to be fixed for any given ramp
+        fit.
+            single of slope variance term:
+                slope_var[Diff.single, :] = ([tau[i] + tau[i+1] - min(t_bar[i], t_bar[i+1]))
+            double of slope variance term:
+                slope_var[Diff.double, :] = ([tau[i] + tau[i+2] - min(t_bar[i], t_bar[i+2]))
```

Comment:
I think you still need to change the slope_var in these comments to var_slope_coeffs.